subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java

```diff
@@ -38,7 +38,7 @@
 public class ClasspathUtil {
     public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathElements) {
         try {
-            Set<URI> original = new HashSet<URI>();
+            Set<URI> original = new HashSet<>();
```

Comment:
We cannot omit type information here as the base-services module needs to be source compatible with Java 6